src/middleware/company.authentication.middleware.ts

```diff
@@ -48,3 +34,26 @@ export const companyAuthentication = async (req: Request, res: Response, next: N
     next(err);
   }
 };
+
+async function processTransaction (entityNumber: string, returnURL: string, req: Request): Promise<[string, string]> {
+  if (req.path.endsWith(`/${RESUME}`)) {
+    const { transactionId } = req.params;
+
+    if (transactionId) {
+      const transactionResource = await getTransaction(req, transactionId);
+      const entityNumberTransaction = transactionResource.companyNumber;
+
+      if (!entityNumber || (entityNumberTransaction && entityNumber !== entityNumberTransaction)) {
+        entityNumber = entityNumberTransaction as string;
+        returnURL = req.originalUrl;
+      }
+      if (entityNumber && transactionResource.status === CLOSED_PENDING_PAYMENT) {
+        returnURL = req.originalUrl;
+      }
+    } else {
+      logger.errorRequest(req, "Invalid transactionId");
+      throw new Error("Invalid transaction");
+    }
+  }
+  return [entityNumber, returnURL];
+}
```

Comment:
Line 28 to 31,  of original:
} else {
      logger.errorRequest(req, "No entity number to authenticate against");
      throw new Error("OE number not found in session");
    }
I think this leads to back button errors after payment completes. 
Should we re-direct to the start of the update journey as that would then force the user to select an OE number?

else {
   logger.infoRequest(req, `No entity number to authenticate against - redirecting to ${const.UPDATE_LANDING_URL)}`);
   return res.redirect(const.UPDATE_LANDING_URL);
}